### ISSUE TITLE
[a11y]: UI Shell [Header & Header with Actions and Right Panel] - notifications SR bug

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#15269]
### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.41.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Package

@carbon/react

### Browser

FF, Safari & iPhone

### Package version

@carbon/react@1.41.1

### Automated testing tool and ruleset

Manual QA

### Assistive technology

VO & JAWS

### Description
Notification button is not announcing as expanded by screen reader

Notification button is visually looking as expanded but screen reader is not announcing it.

### Recommended Fix
Screen reader should anounce as 'expanded'.

People who uses screen reader will not be aware of the state of the element.

### WCAG 2.1 Violation
4.1.2 Name, Role, Value
